src/metabase/driver/sql_jdbc/execute.clj

```diff
@@ -339,10 +345,22 @@
    options          :- ConnectionOptions
    f                :- fn?]
   (binding [*connection-recursion-depth* (inc *connection-recursion-depth*)]
```

Comment:
We spoke about the need for these binding if spec is passed, would be interested in poking that a bit in case we can drop these recursion vars and just rely on *db-conn* (maybe could rely on `existing-connection (or (:connection db-or-id-or-spec) *db-conn* )`